### PR TITLE
Expose blocktime

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/main/java/com/iexec/blockchain/config/ConfigExposureController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public ConfigExposureController(ChainConfig chainConfig) {
+        this.chainConfig = chainConfig;
+    }
+
+    /**
+     * Unauthenticated endpoint.
+     */
+    @GetMapping("/blocktime")
+    public ResponseEntity<Integer> getBlockTime() {
+        return ResponseEntity.ok(chainConfig.getBlockTime());
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply src/main/java/com/iexec/blockchain/config/ConfigExposureController.java:39]: In this case the endpoint would be something like `/config/chain`.

[the remark this answers is not in the document]
Comment (mcornaton) [reply src/main/java/com/iexec/blockchain/config/ConfigExposureController.java:39]: That was an option I considered, but:
- I wanted to stay as simple as possible as of now;
- we already have a `ChainConfig` class in `iexec-blockchain-adapter-api` so we'd need to find another eloquent naming. Something like `ChainPublicConfig`?

[the remark this answers is not in the document]
Comment (jeremyjams) [reply src/main/java/com/iexec/blockchain/config/ConfigExposureController.java:39]: `PublicChainConfig` and `/config/chain` is fine too me.

[the remark this answers is not in the document]
Comment (mcornaton) [reply src/main/java/com/iexec/blockchain/config/ConfigExposureController.java]: Fixed :)

File: src/main/java/com/iexec/blockchain/config/ConfigExposureController.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     * Unauthenticated endpoint.
+     */
+    @GetMapping("/chain")
+    public ResponseEntity<PublicChainConfig> getPublicChainConfig() {
+        final Integer blockTime = chainConfig.getBlockTime();
+        final PublicChainConfig publicChainConfig = PublicChainConfig
+                .builder()
+                .blockTime(Duration.ofSeconds(blockTime))
+                .build();
+        log.info("Sending public chain config [{}]", publicChainConfig);
```
Comment (jeremyjams) [inline src/main/java/com/iexec/blockchain/config/ConfigExposureController.java:49]: I think you can remove that log.

**Comment by jeremyjams** [comment]
You can validate block time in config following JSR380 spec, see https://github.com/iExecBlockchainComputing/iexec-sms/blob/a95cc1dd46a105dc47d3b4fc42bb674bdf03eee3/src/main/java/com/iexec/sms/tee/workflow/TeeWorkflowConfiguration.java#L69-L69 for eg.

File: build.gradle [hunk 3]
```diff
@@ -46,6 +46,7 @@ dependencies {
 	implementation "io.springfox:springfox-swagger-ui:3.0.0"
 	implementation 'org.springframework.boot:spring-boot-starter-security'
 	implementation 'org.springframework.boot:spring-boot-starter-data-mongodb'
+	implementation 'org.springframework.boot:spring-boot-starter-validation'
```
Comment (jeremyjams) [inline build.gradle:49]: Isn't javax.validation:validation-api enough?

Comment (mcornaton) [reply build.gradle:49]: Sadly, no. We get the following error:
```
Parameter 0 of constructor in com.iexec.blockchain.tool.ChainConfig required a bean of type 'javax.validation.Validator' that could not be found.
```

Comment (jeremyjams) [reply build.gradle:49]: I see thanks

File: src/main/java/com/iexec/blockchain/config/ConfigExposureController.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.time.Duration;
+
+@RestController
+@RequestMapping("/config")
+@Slf4j
```
Comment (jeremyjams) [inline src/main/java/com/iexec/blockchain/config/ConfigExposureController.java:31]: Useless

File: src/main/java/com/iexec/blockchain/config/ConfigExposureController.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+public class ConfigExposureController {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/blockchain/config/ConfigExposureController.java:32]: ConfigExposureController -> PublicConfigurationController

**Review by jeremyjams — APPROVED** [review]
Thanks :+1:

**Review by zguesmi — APPROVED** [review]

**Pull request merged**